dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,176 @@
+import math
+import numpy as np
+from sklearn.base import clone
+from sklearn.model_selection import ParameterSampler
+
+from toolz import groupby
+from distributed import as_completed, default_client
+from dask_searchcv.model_selection import DaskBaseSearchCV
+
+
+def _get_nr(R, eta=3):
+    s_max = math.floor(math.log(R, eta))
+    B = (s_max + 1) * R
+
+    brackets = list(reversed(range(int(s_max + 1))))
+    N = [math.ceil(B / R * eta**s / (s + 1)) for s in brackets]
+    R = [int(R * eta**-s) for s in brackets]
+    return list(map(int, N)), list(map(int, R)), brackets
+
+
+def _partial_fit(model_and_meta, x, y, meta=None, fit_params={},
+                 dry_run=False):
+    model, m = model_and_meta
+    if meta is None:
+        meta = m
+    while meta['iterations'] < meta['partial_fit_calls']:
+        if not dry_run:
+            model.partial_fit(x, y, **fit_params)
```

Comment:
I would have expected this to iterate through a few different x's and y's

Reply:
I think we should do this; it's supported by [the paper][1] in section 4.1 (one `partial_fit` call does not see the entire dataset).

[1]:https://arxiv.org/pdf/1603.06560.pdf

I think by default I'll call `partial_fit` with each chunk, and give some keyword arg to specify how many chunks to pass to `partial_fit`.

Reply:
Ideally this function would not consume dask arrays (this would be nested execution).  Ideally it would consume a numpy array or a list of numpy arrays.

Reply:
So we'll probably need to do the logic of passing chunks of our data in the client process.  I suggest that we take in dask arrays and convert them to delayed objects:

```python
def _hyperband(..., x:da.Array, y:da.Array, ...):
    x_blocks = x.to_delayed()
    y_blocks = y.to_delayed()

    client.submit(partial_fit, model_future, x_blocks[i], y_blocks[i])
    # or 
    client.submit(partial_fit, model_future, x_blocks[i:i+k], y_blocks[i:i+k])
```

Reply:
Thanks for this. I've implemented this, and tested it to make sure it's working.